www/js/app.js

```diff
@@ -2236,9 +2236,10 @@ function filterClickEvent (event) {
         }
         // DEV: The href returned below is the href as written in the HTML, which may be relative
         var href = clickedAnchor.getAttribute('href');
-        // We assume that, if an absolute http(s) link is hardcoded inside an HTML string, it means it's a link to an external website.
-        // We also do it for ftp even if it's not supported any more by recent browsers...
-        if (/^(?:http|ftp)/i.test(href)) {
+        // We assume that, if an absolute http(s) link is hardcoded inside an HTML string, it means it's a link to an external website
+        // (this assumption is only safe for non-Replay archives, but we deal with those separately above: they are routed to handleClickOnReplayLink).
+        // Additionally, by comparing the protocols, we can filter out protocols such as `mailto:`, `tel:`, `skype:`, etc. (these should open in a new window).
+        if (/^(?:http|ftp)/i.test(href) || clickedAnchor.protocol && clickedAnchor.protocol !== ':' && articleWindow.location.protocol !== clickedAnchor.protocol) {
```

Comment:
Technically `articleWindow.location.protocol` will never be `ftp:` right? So you can probably remove that from the regex to make the condition more self explanatory.

Also what links have a protocol of `:`?

Reply:
You're right, that's old code that's only really there to show a dev how to test for different protocols, but it's a bit redundant with the following `||` test. We do, however, need to keep the `^http` test here because we shouldn't normally find absolute `^http`-prefixed protocols for a ZIM link except in the case of Replay-based ZIMs (where wombat.js is in charge of overriding these and transforming them to ZIM links, but only on-the-fly). So, if we do find an `http`-prefixed protocol it is an external link and has to be opened outside the sandbox.

Regarding `:`, I was surprised myself to find these, but I was forced to put in this condition. It seems to be an edge case in Chromium browsers, probably a bug t.b.h., where some internal code has stripped the protocol because the URL resolves to a local URL that is same-origin, but the colon has been left in whatever search-and-replace operation happens internally. Browser quirks are the bane of my life....

I'll amend.

Reply:
As a random guess, I thought maybe relative links would have an "empty" protocol (the MDN docs say that the protocol always ends with `:`). But that's not the case in Chrome:

![2024-07-09_11-10](https://github.com/kiwix/kiwix-js/assets/57832/e5728c3b-3d9b-4b67-96e1-5c4bdbca85da)

Reply:
I'll see if I can find the condition and browser that produced this... But I was definitely forced to add it at some point... I should have documented it I guess.

Reply:
So with some quick testing I haven't yet identified anything, but I'm reluctant to remove it because I know I was forced to put it in the code, and it could be some quirk in Edge Legacy or another browser I can't test right now. Or it could be a side-effect of wombat.js processing of links from when this function originally tried to cover that case and before I added in specific processing of Replay links...

What I'll do is add a DEV note to say that the usefulness of the condition should be evaluated and if no longer necessary after careful testing it can be removed.